src/controllers/expense.controller.js

```diff
@@ -0,0 +1,112 @@
+const expenseService = require('./../services/expense.service');
+const userService = require('./../services/user.service');
+
+const getExpenses = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
+
+  try {
+    const expenses = await expenseService.getAllExpense({
+      userId,
+      categories,
+      from,
+      to,
+    });
+
+    res.status(200).send(expenses);
+  } catch (error) {
+    res.status(500).send({ message: 'Server error' });
+  }
+};
+
+const getOneExpense = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const expense = await expenseService.getExpenseById(id);
+
+    if (!expense) {
+      return res.status(404).send({ message: 'Expense not found' });
+    }
+    res.status(200).send(expense);
+  } catch (error) {
+    res.status(500).send({ message: 'Server error' });
+  }
+};
+
+const createExpense = async (req, res) => {
+  const data = req.body;
+
+  if (
+    !data.userId ||
+    !data.title ||
+    !typeof data.categories ||
+    !typeof data.spentAt ||
+    !typeof data.amount
```

Comment:
The validation logic for `categories`, `spentAt`, and `amount` is incorrect. The `typeof` operator returns a string, so you should compare it to the expected type, e.g., `typeof data.categories !== 'object'`. This ensures that the fields are of the correct type.